File: Dynamic Programming/198_house_robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int one = 0, two = nums[0];

        for(int i = 1; i < nums.size(); i += 1) {
            int temp = two;
            two = max(two, nums[i] + one);
            one = temp;
        }

        return two;
    }
};

// revised - 02/03/2025

File: Dynamic Programming/198_house_robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 1:
            return nums[0]

        nums.insert(0, 0)
        for i in range(2, n + 1):
            nums[i] = max(nums[i - 1], nums[i] + nums[i - 2])

        return nums[n]